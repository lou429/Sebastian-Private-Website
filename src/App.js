import './App.scss';
import {BrowserRouter as Router, Routes, Route} from 'react-router-dom';
import NavBar from './components/navbar.js'
import Dev from "./components/dev.js";
import Home from "./components/home.js";
import SignInPage from './components/SignInPage.js';
import User from "./components/UserPage.js";
import NoPage from "./components/NoRouteFound.js"
import Circle from "./components/Circle.js";
import './style.scss';

function App() {
    return (
        <div className="page-wrapper">
            {/* Randomly drawn circles */}
            < Circle/>

            <div className="glass-page-wrapper">
                <NavBar/>

                <div className="content-wrapper">
                    {/* React router nav pages */}
                    <Router>
                            <Routes>
                                <Route path='*' exact element={< NoPage />}/>
                                <Route path="/" exact element={< Home />}/>
                                <Route path="/home" element={< Home />}/>
                                <Route path="/dev" element={< Dev />}/>
                                <Route path="/SignIn" element={< SignInPage />}/>
                                <Route path="/User" element={< User />}/>
                            </Routes>
                        </Router>
                </div>
            </div>
        </div>
    );
}

export default App;