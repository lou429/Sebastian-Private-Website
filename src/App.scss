@import url('https://fonts.googleapis.com/css2?family=Montserrat:wght@200&display=swap');

$container-background-color-main: rgb(87, 87, 87);

.App {
  text-align: center;
}

.base-content-container {
  margin: 0px 70px;
  background-color: $container-background-color-main;
  border: thin 1px black;
  min-height: 100%;
  left: 10px;
  right: 10px;
}

.base-content-row {
  display: inline-block;
  margin: 0px auto;
  text-align: center;
  font-family: 'Montserrat', sans-serif;
}

#github-profile {
  
  margin: 10px 0px;
}