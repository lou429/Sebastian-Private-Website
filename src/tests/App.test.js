import { render } from '@testing-library/react';
import App from '../App';

test('Render full app', () => {
  const { getByText } = render(<App/>);
  expect(getByText('Home')).toBeInTheDocument();
});
