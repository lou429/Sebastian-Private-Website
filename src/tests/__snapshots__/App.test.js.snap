// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<div>
  <div
    class="page-wrapper"
  >
    <div
      class="circle-container"
    >
      <div
        class="animate__animated animate__fadeInUpBig animate__slower"
        id="circle-0"
        style="height: 23rem; width: 23rem; right: 38%; bottom: 10%;"
      />
      <div
        class="animate__animated animate__fadeInDownBig animate__slower"
        id="circle-1"
        style="height: 22rem; width: 22rem; left: 7%; top: 3%;"
      />
      <div
        class="animate__animated animate__fadeInUpBig animate__slower"
        id="circle-2"
        style="height: 18rem; width: 18rem; right: 27%; bottom: 13%;"
      />
      <div
        class="animate__animated animate__fadeInUpBig animate__slower"
        id="circle-3"
        style="height: 11rem; width: 11rem; bottom: 3%; right: 24%;"
      />
      <div
        class="animate__animated animate__fadeInUpBig animate__slower"
        id="circle-4"
        style="height: 23rem; width: 23rem; left: 36%; bottom: 6%;"
      />
    </div>
    <div
      class="glass-page-wrapper"
    >
      <div
        class="nav-wrapper"
      >
        <div
          class="nav-heading"
        >
          <div
            class="navbar-heading-container"
          >
            <div
              class="navbar-heading-row"
            >
              <a
                href="https://github.com/lou429"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  alt="Github profile"
                  height="120px"
                  id="github-profile"
                  src="https://github.com/lou429.png"
                  width="120px"
                />
              </a>
            </div>
          </div>
        </div>
        <div
          class="nav-body"
        >
          <div
            class="nav-body-container"
          >
            <a
              class="nav-link-active"
              href="/"
            >
              Home
            </a>
            <a
              class="nav-link"
              href="/dev"
            >
              Dev page
            </a>
            <a
              class="nav-link"
              href="/user"
            >
              About me
            </a>
            <a
              class="nav-link"
              href="/contact"
            >
              Contact me
            </a>
          </div>
        </div>
        <div
          class="nav-footer"
        >
          <div
            class="navbar-footer-container"
          >
            <h1
              class="animate__animated animate__fadeInLeft animate_slower animation__delay 1.5s"
              id="footerText"
            />
          </div>
        </div>
      </div>
      <div
        class="content-wrapper"
      >
        <div
          class="content-container"
        >
          <div
            class="content-heading"
          >
            <h1>
              Home page
            </h1>
          </div>
          <div
            class="content-body"
          >
            <h6>
              This website is a work in progress therefor it is not fully finished.
            </h6>
            <br />
            <h6>
              Most of the core features of this web app work as intended but bugs are to be expected
            </h6>
            <br />
            <h6>
              If you encounter any bugs please send a message on the contact page alongside with what you were doing that caused
            </h6>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
